scrapers/scrape_zg.py

```diff
@@ -1,21 +1,46 @@
 #!/usr/bin/env python3
 
+import re
+from io import StringIO
+from urllib.parse import urlparse
+from bs4 import BeautifulSoup
+import pandas as pd
 import scrape_common as sc
 
 print('ZG')
 d = sc.download('https://www.zg.ch/behoerden/gesundheitsdirektion/amt-fuer-gesundheit/corona')
 sc.timestamp()
+soup = BeautifulSoup(d, 'html.parser')
 d = sc.filter(r'Infizierte Personen|Genesene Personen|Verstorbene Personen|Stand:', d)
+print('Date and time:', sc.find(r'Stand:? ([^<]+ Uhr)<', d))
 
-# 2020-03-23
-"""
+detailed_stats = soup.find("a", text=re.compile("Detaillierte Statistik"))
+if detailed_stats:
+    d_detailed_stats = sc.download(detailed_stats['href'])
+    d_detailed_stats_comps = urlparse(detailed_stats['href'])
+    csv_path = sc.find(r'csv_path:"([^"]+)"', d_detailed_stats)
+    csv_url = f"{d_detailed_stats_comps.scheme}://{d_detailed_stats_comps.netloc}{csv_path}"
```

Comment:
Just use `https://www.zg.ch`  There is no need for `urlparse` at all.

Reply:
Changed